internal/services/access_flow.go

```diff
@@ -0,0 +1,476 @@
+				Type:  types.StringValue("tag"),
+				Name:  types.StringValue(tag.Name),
+				Value: types.StringValue(tag.Value),
+			})
+		}
+	}
+
+	return filters
+}
+
+func convertTriggerToApiModel(trigger models.Trigger) (*apono.AccessFlowTriggerV1, diag.Diagnostics) {
+	var data apono.AccessFlowTriggerV1
+	data.Type = trigger.Type.ValueString()
+
+	if trigger.Timeframe != nil {
+		var timeframeDays []apono.DayOfWeek
+		for _, day := range trigger.Timeframe.DaysInWeek.Elements() {
+			timeframeDays = append(timeframeDays, apono.DayOfWeek(utils.AttrValueToString(day)))
+		}
+
+		startOfDayTimeInSeconds, err := utils.DayTimeFormatToSeconds(trigger.Timeframe.StartTime.ValueString())
+		if err != nil {
+			diagnostics := diag.Diagnostics{}
+			diagnostics.AddError("Client Error", fmt.Sprintf("Failed to parse start time: %s", trigger.Timeframe.StartTime.ValueString()))
+			return nil, diagnostics
+		}
+		endOfDayTimeInSeconds, err := utils.DayTimeFormatToSeconds(trigger.Timeframe.EndTime.ValueString())
+		if err != nil {
+			diagnostics := diag.Diagnostics{}
+			diagnostics.AddError("Client Error", fmt.Sprintf("Failed to parse end time: %s", trigger.Timeframe.EndTime.ValueString()))
+			return nil, diagnostics
+		}
+
+		dataTimeFrame := apono.AccessFlowTriggerV1Timeframe{
+			StartOfDayTimeInSeconds: startOfDayTimeInSeconds,
+			EndOfDayTimeInSeconds:   endOfDayTimeInSeconds,
+			DaysInWeek:              timeframeDays,
+			TimeZone:                trigger.Timeframe.TimeZone.ValueString(),
+		}
+
+		data.Timeframe = *apono.NewNullableAccessFlowTriggerV1Timeframe(&dataTimeFrame)
+	}
+
+	return &data, nil
+}
+
+func getIdentitiesIdsByNameAndType(ctx context.Context, identityName string, identityType string, availableIdentities []apono.IdentityModel2, aponoClient *apono.APIClient) ([]string, diag.Diagnostics) {
+	switch identityType {
+	case "user":
+		user, _, err := aponoClient.UsersApi.GetUser(ctx, identityName).Execute()
+		if err != nil {
+			return nil, utils.GetDiagnosticsForApiError(err, "get", "user", identityName)
+		}
+
+		return []string{user.GetId()}, nil
+
+	case "group", "context_attribute":
+		var identitiesIds []string
+		for _, identity := range availableIdentities {
+			if identity.Name == identityName && strings.ToLower(identity.Type) == identityType {
+				identitiesIds = append(identitiesIds, identity.Id)
+			}
+		}
+		if len(identitiesIds) == 0 {
+			diagnostics := diag.Diagnostics{}
+			diagnostics.AddError("Client Error", fmt.Sprintf("Failed to get identity %s: %s", identityType, identityName))
+			return nil, diagnostics
+		}
+
+		return identitiesIds, nil
+
+	default:
+		diagnostics := diag.Diagnostics{}
+		diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unsupported indentity type: %s, please update the provider to support this type or removed it using UI/API", identityType),
+		)
+		return nil, diagnostics
+	}
+}
+
+func getUniqueListOfIdentities(identities []models.Identity) []models.Identity {
```

Comment:
what is the use for this method?

Reply:
Remove duplicate Identities from list, because when you get 2 group with the same name in the API, we need to convert it to single object in Terraform

Reply:
add comment